source/simulator.hpp

```diff
@@ -49,6 +51,7 @@ class Simulator {
     std::unordered_set<std::shared_ptr<ISwitch>> m_switches;
     std::unordered_set<std::shared_ptr<IConnection>> m_connections;
     std::unordered_set<std::shared_ptr<ILink>> m_links;
+    std::shared_ptr<Scenario> m_scenario;
```

Comment:
I think we do net need any pointer here because `Scenario` is a concrete object, not abstract
```suggestion
    Scenario m_scenario;
```